atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImplTest.java

```diff
@@ -99,20 +99,6 @@ public void valueCacheCreatesValidatingTransactionCaches() {
         assertThat(getRemotelyReadCells(scopedCache, TABLE, CELL_1)).containsExactlyInAnyOrder(CELL_1);
     }
 
-    @Test
-    public void updateCacheOnCommitFlushesValuesToLatestSequenceCache() {
```

Comment:
The semantics have changed:
Before: if transactions A, B start at the same time, but A finishes before B even gets its transaction cache, then B will see the reads from A.
After: A and B start at the same time; if A finishes, then B will still have the same cache that A started with, _unless_ C starts after A finishes but with the same sequence, in which case B and C will have the same cache.

Effectively it is a very minor difference in when a snapshot for a fixed sequence is re-updated, and is only really relevant in cases where there are _very_ few updates (and hence the sequence isn't changing much at all). In these cases, the new semantics are marginally less efficient.

Reply:
Yeah, that makes sense, it's a corner case anyway and the difference is minor